### PR TITLE
Some minor performance enhancements

### PR DESCRIPTION
**Pull request by midzer**
Hi.
I ran the source code through latest cppcheck and found few parts where some optimizations are possible.

Project built nicely on Qt 5.5.1 under Debian Jessie.

### CODE REVIEW

File: librecad/src/lib/gui/rs_painter.h [hunk 1]
```diff
@@ -50,9 +50,8 @@ class QBrush;
  */
 class RS_Painter {
 public:
-    RS_Painter() {
+    RS_Painter(): offset(RS_Vector(0.0,0.0)) {
```
Comment (dxli) [inline librecad/src/lib/gui/rs_painter.h]: @midzer 

```C++
RS_Painter(): offset{0.0,0.0}) {
```

**Comment by dxli** [comment]
@midzer 

sorry for being late in reply :( I was a little bit busy lately

Yes, found an ugly part in LC, and thanks a lot :)

we really want continuous code review, and hopefully upgrade our legacy codebase in the process.

File: librecad/src/lib/gui/rs_painter.h (hunk 1 above)
Comment (midzer) [inline librecad/src/lib/gui/rs_painter.h]: i dont know {} syntax to initialize a variable tbh

Comment (dxli) [inline librecad/src/lib/gui/rs_painter.h]: try to use aggregate initialization, even though for most time, it doesn't matter (the case here).

interesting difference:

```C++
struct A;
A a(); // declare a function A a() -> A {...}
A b{}; // define a variable a by default ctor of A
vector<int> v(2, 1); // v contains {1, 1}
vector<int> w{2, 1}; // w contains {2, 1}
```

Comment (midzer) [inline librecad/src/lib/gui/rs_painter.h]: thanks for clarification!
i did not get offset being an array :)

**Comment by dxli** [comment]
@midzer 

please update your branch there, so we can quickly merge and close this one :+1:

**Comment by midzer** [comment]
@dxli what do you mean by updating my branch? doing a fetch upstream & merge on my patch1 branch to update to head?

**Comment by dxli** [comment]
@midzer 

Thanks

**Pull request merged**